tests/unittests/GradCheckTest.cpp

```diff
@@ -149,49 +150,19 @@ void performGradCheck(ExecutionEngine &EE, Context &ctx, SaveNode *result,
 
     auto numericGrad = (plusLoss - minusLoss) / (2 * delta);
     auto analyticalGrad = analyticalGradsH.raw(i);
-
     auto err = gradDiff(analyticalGrad, numericGrad);
 
+    if (numericGrad != 0.0) {
+      allZeroGradients = false;
+    }
     // Make sure that the analytical and numerical gradients agree.
     EXPECT_LE(err, allowedError);
   }
-}
 
-TEST_P(InterpreterGrad, gradientCheckFCConcatRELU) {
```

Comment:
I don't agree with removal of this test. This is the only test which executes gradient of Concat. Maybe the name is unfortunate? Do you see 0 gradients here even when you replace `initXavier()` by `randomize()`?

FYI, we have multiple `createFullyConnected()` constructors. Some of which accept W and B as arguments, some create them as hidden (but still accessible) vars. If you prefer to have W and B as constants, feel free to create&init them in test body and call this method of `F`:
https://github.com/pytorch/glow/blob/259111daf2def53c6d74dbd2d2d5e988bc5a9d3d/lib/Graph/Graph.cpp#L647

Reply:
Thanks. I've added gradientCheckConcat() to address this concern. To me gradientCheckFCConcatRELU is more like an integration test that shouldn't be here and reasoning about it very hard. gradientCheckConcat is trying to use as little operations as possible in addition to concat to minimize its complexity. Let me know what you think!

Reply:
Looks good!